src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  static removeFromAliveList(animal) {
+    const index = Animal.alive.findIndex(a => a === animal);
+    if (index !== -1) {
+      Animal.alive.splice(index, 1);
+    }
+  }
+
+  takeDamage(damage) {
+    if (this.health > 0) {
+      this.health -= damage;
+      if (this.health <= 0) {
+        Animal.removeFromAliveList(this);
+      }
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
```

Comment:
Add constructor(name, health).
Constructor constructor(name, health): The Herbivore constructor must accept two parameters - name (name of the animal) and health (health of the animal). This constructor first calls the constructor of its parent Animal class, passing it name and health. This is done using super(name, health);. Thus, the Herbivore class inherits the properties and methods of the Animal class.